tasks/sep/fp-recovery/005-set-game-implementation/justfile

```diff
@@ -0,0 +1,91 @@
+  #!/usr/bin/env bash
+  set -euo pipefail
+  mkdir out
+  sed -e 's/<L2_CHAIN>/{{l2}}/g' -e 's/<L1_CHAIN>/{{l1}}/g' templates/README.md > out/README.md
+  cp templates/input.json out/input.json
+  cp templates/VALIDATION.md out/VALIDATION.md
+  cp templates/NestedSignFromJson.s.sol out/NestedSignFromJson.s.sol
+
+  if [[ "{{l1}}" == "sepolia" ]]; then
+    cat > out/.env << EOF
+  ETH_RPC_URL="https://ethereum-sepolia.publicnode.com"
+  COUNCIL_SAFE=0xf64bc17485f0B4Ea5F06A96514182FC4cB561977
+  FOUNDATION_SAFE=0xDEe57160aAfCF04c34C887B5962D0a69676d3C8B
+  EOF
+  elif [[ "{{l1}}" == "mainnet" ]]; then
+    cat > out/.env << EOF
+  ETH_RPC_URL="https://ethereum.publicnode.com"
+  COUNCIL_SAFE=0xc2819DC788505Aac350142A7A707BF9D03E3Bd03
+  FOUNDATION_SAFE=0x847B5c174615B1B7fDF770882256e2D3E95b9D92
+  EOF
+  else
+    echo "Unknown L1 chain."
+    exit 1
+  fi
+
+  CONFIG_URL="https://raw.githubusercontent.com/ethereum-optimism/superchain-registry/refs/heads/main/superchain/configs/{{l1}}/{{l2}}.toml"
+  CONFIG_TOML=$(curl -s --show-error --fail "${CONFIG_URL}")
+  SYSTEM_CONFIG=$(echo "${CONFIG_TOML}" | yq -p toml .addresses.SystemConfigProxy)
+  OWNER_SAFE=$(echo "${CONFIG_TOML}" | yq -p toml .addresses.ProxyAdminOwner)
+
+  cat >> out/.env << EOF
+  SAFE_NONCE=""
+  L1_CHAIN_NAME="{{l1}}"
+  L2_CHAIN_NAME="{{l2}}"
+  SYSTEM_CONFIG="${SYSTEM_CONFIG}"
+  OWNER_SAFE="${OWNER_SAFE}"
+  EOF
+
+# Generate the `input.json` with the game type and implementation to set.
+set-implementation gameType='' implAddr='':
+  #!/usr/bin/env bash
+  set -euo pipefail
+  . out/.env
+  export ETH_RPC_URL # Make available to cast since the .env file is in the out subdir.
+  SET_GAME_TYPE_SIG="setImplementation(uint32,address)"
+  ENCODED_CALL=$(cast calldata "${SET_GAME_TYPE_SIG}" "{{gameType}}" "{{implAddr}}")
+
+  DGF_ADDR=$(cast call "${SYSTEM_CONFIG}" "disputeGameFactory()(address)")
+
+  TX_DESC="Sets the implementation for game type {{gameType}} to {{implAddr}} in the \`DisputeGameFactory\`."
+  TX=$(jq --arg desc "${TX_DESC}" "(.metadata.description = \$desc) |
+                (.data = \"$ENCODED_CALL\") |
+                (.to = \"${DGF_ADDR}\") |
+                (.contractInputsValues._gameType = \"{{gameType}}\") |
+                (.contractInputsValues._impl = \"{{implAddr}}\")" ./templates/tx/setImplementation.json)
+  jq --argjson tx "${TX}" --arg desc "${TX_DESC} " '(.metadata.description += $desc) | (.transactions += [$tx])' ./out/input.json > ./out/input.tmp.json
+  mv ./out/input.tmp.json ./out/input.json
+
+  # Add a precheck for this implementation change
+  sed -I "" -e 's|// INSERT NEW PRE CHECKS HERE|_precheckDisputeGameImplementation(GameType.wrap({{gameType}}), {{implAddr}});\n        // INSERT NEW PRE CHECKS HERE|' out/NestedSignFromJson.s.sol
+  sed -I "" -e 's|// INSERT NEW POST CHECKS HERE|_checkDisputeGameImplementation(GameType.wrap({{gameType}}), {{implAddr}});\n        // INSERT NEW POST CHECKS HERE|' out/NestedSignFromJson.s.sol
```

Comment:
same cross-platform `sed` question

Reply:
Yeah this one had to be adjusted - I couldn't remember if `-I` or `-i` was the standard option - turns out it was the lowercase one.

Reply:
Upon further testing there's differences in which temporary files get left around so I gave up on having sed edit in place.